build.gradle

```diff
@@ -237,6 +239,7 @@ project(':cruise-control') {
     }
     compile project(':cruise-control-metrics-reporter')
     compile project(':cruise-control-core')
+    compile project(':cruise-control-util')
```

Comment:
Now that we have `cruise-control-util`, should we extract the shared logic from `metrics-reporter` to `util` and drop `compile project(':cruise-control-metrics-reporter')`?

Reply:
We thought it through with @egyedt and we think it might not be possible to completely drop this dependency. The reason is most of the classes are used from the `cruise-control` module and some of them would look a bit weird in a util module (for instance `BrokerMetric` or `TopicMetric`). 
It might be a bit better to rename this util module to `cruise-control-metrics-reporter-core` (which might be better aligned with the classes inside it), move the common classes there and just depend on that in both `cruise-control` and `cruise-control-metrics-reporter`.
The drawback of this solution is that it wouldn't be too different than just having only `cruise-control-metrics-reporter`.

Also parts of `CruiseControlMetricsReporterConfig` are used in `cruise-control` (checking reporting interval against sampling interval) which is quite hard to refactor as it requires the reporting config's name. Is this check required on Cruise Control's side?

For these reasons I think for now we could just go with the original solution that is move `EnvConfigProvider` to `cruise-control-metrics-reporter` and consider the utils/core module in a later PR. What do you think?

Reply:
Thanks for detailed explanation!

> For these reasons I think for now we could just go with the original solution that is move `EnvConfigProvider` to `cruise-control-metrics-reporter` and consider the utils/core module in a later PR. What do you think?

I agree with (1) using the original solution and (2) leaving the utils/core module to a later PR.